tests/test_generation_utils.py

```diff
@@ -1680,3 +1700,28 @@ def test_max_new_tokens_decoder_only(self):
         # max_new_tokens and max_length serve the same purpose and should not be used together.
         with self.assertWarns(UserWarning):
             gpt2_model.generate(decoder_input_ids=input_ids, max_new_tokens=10, max_length=20)
+
+    def test_encoder_decoder_generate_with_inputs_embeds(self):
+        article = """Justin Timberlake and Jessica Biel, welcome to parenthood."""
+        tokenizer = BartTokenizer.from_pretrained("hf-internal-testing/tiny-random-bart")
+        model = BartForConditionalGeneration.from_pretrained("hf-internal-testing/tiny-random-bart", max_length=5).to(
+            torch_device
+        )
+        model.config.eos_token_id = None
+        input_ids = tokenizer(article, return_tensors="pt").input_ids.to(torch_device)
+        inputs_embeds = model.get_input_embeddings()(input_ids)
+
+        output_sequences = model.generate(inputs_embeds=inputs_embeds)
+
+        self.assertEqual(output_sequences.shape, (1, 5))
+
+    def test_decoder_generate_with_inputs_embeds(self):
+        article = """I need input_ids to generate"""
+        tokenizer = GPT2Tokenizer.from_pretrained("hf-internal-testing/tiny-random-gpt2")
+        model = GPT2LMHeadModel.from_pretrained("hf-internal-testing/tiny-random-gpt2", max_length=5).to(torch_device)
+        input_ids = tokenizer(article, return_tensors="pt").input_ids.to(torch_device)
+        inputs_embeds = model.get_input_embeddings()(input_ids)
+
+        # cannot generate from `inputs_embeds` for decoder only
+        with pytest.raises(ValueError):
```

Comment:
Decoder-only can't generate with `inputs_embeds` since predicted ids are append to starting ids => so it's assumed that starting ids are `input_ids` and not `inputs_embeds`

Reply:
With not use `self.assertRaises(ValueError)` here?